tracer/test/Datadog.Trace.ClrProfiler.Managed.Tests/AutoInstrumentation/AWS/MemoryStreamCarrierGetter.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="MemoryStreamCarrierGetter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Amazon.SimpleNotificationService.Model;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNSTests;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Propagators;
+using Newtonsoft.Json;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNSTests
+{
+    public struct MemoryStreamCarrierGetter : ICarrierGetter<PublishRequestProxy>
+    {
+        private const string DatadogKey = "_datadog";
+        private const string CacheKey = "TestCacheKey";  // constant key
+        private static readonly Dictionary<string, Dictionary<string, string>> Cache = new();
+
+        public IEnumerable<string> Get(PublishRequestProxy carrier, string key)
+        {
+            object attributeValue;
+
+            if (!Cache.TryGetValue(CacheKey, out var cachedData) && carrier.MessageAttributes.TryGetValue(DatadogKey, out attributeValue))
+            {
+                var messageAttributeValue = attributeValue as MessageAttributeValue;
+
+                if (messageAttributeValue != null && messageAttributeValue.BinaryValue is MemoryStream memoryStream)
+                {
+                    memoryStream.Position = 0;
+                    using var reader = new StreamReader(memoryStream, Encoding.UTF8);
```

Comment:
I figured I'd try this out on .NET Framework (I have a Windows machine) and it looks like you'll need the following API with all required arguments to get this to work on all platforms:

```suggestion
                    using var reader = new StreamReader(memoryStream, encoding: Encoding.UTF8, detectEncodingFromByteOrderMarks: true, bufferSize: -1, leaveOpen: true);
```

Reply:
derp,  just realized .net framwork is just for windows 💡 🤦  all these terms are still fuzzy to me

Reply:
No worries! But yeah I wanted to quickly validate that the unit tests for .NET Framework (windows-only runtime) worked, and that's how I spotted some of these small issues